acl2_kernel/kernel.py

```diff
@@ -53,6 +53,12 @@ def do_execute(self, code, silent, store_history=True, user_expressions=None, al
         try:
             cmd = re.sub(r'[\r\n]|;[^\r\n]*[\r\n]+', ' ', code.strip())
             output = self.acl2wrapper.run_command(cmd, timeout=None)
+            while True:
+                try:
+                    more_output = self.acl2wrapper.run_command(';', timeout=0)
```

Comment:
Thank you for a nice change! One question, you set 0s the keyword argument `timeout`.
It means that the command throws TIMEOUT immediately because the program waits for 0s.
It is too short, isn't it? Would you mind explaining to me why you set it so?

Reply:
Even with a timeout of 0, the underlying OS code will check whether there is further output from ACL2 _**before**_ checking for timeout (I'm certain this is true for any reasonable implementation of read-with-timeout). Thereby, a timeout of 0 means that minimum time is wasted if there is no further output from ACL2.

This does assume that for cases like the output of:

```lisp
(acl2s-defaults :set testing-enabled t)
```

ACL2 does manage to send any further output to the kernel before the next call to `self.acl2wrapper.run_command(';', timeout=0)` but I think this is false only for very rare cases (e.g. if you're building an ACL2 binary image) and these rare cases don't normally occur in notebooks.

Reply:
I see. So you mean the order of the pexpect process is;

1. Check the output,
2. and Measure waiting time.

I am also understand the exceptional situation. Thank you.